src/api/app/models/project.rb

```diff
@@ -968,6 +976,44 @@ def branch_to_repositories_from(project, pkg_to_enable, opts = {})
     end
   end
 
+  def branch_remote_repositories(project)
+    remote_project = RemoteProject.new(name: project)
+    remote_project_meta = Nokogiri::XML(remote_project.meta.to_s)
+    local_project_meta = Nokogiri::XML(to_axml)
+
+    remote_repositories = remote_project.repositories_from_meta
+    remote_repositories -= repositories.where(name: remote_repositories).pluck(:name)
+
+    remote_repositories.each do |repository|
```

Comment:
The main point is that you need to anyway parse the meta xml of the remote project so the easiest way is to just take it and append it to the new project. Otherwise you have to filter out the data from the remote meta and based on this data you have to create the objects, so I see some benefits in this solution. Also with this solution ``update_from_xml!`` and the ``BranchPackage`` model would take care of the error/exception handling.